waspc/data/Generator/templates/server/src/auth/providers/oauth/createRouter.ts

```diff
@@ -49,40 +50,45 @@ export function createRouter(provider: ProviderConfig, initData: { passportStrat
           const getUserFields = () => getUserFieldsFn ? getUserFieldsFn(contextWithUserEntity, { profile: providerProfile }) : Promise.resolve({});
           // TODO: In the future we could make this configurable, possibly associating an external account
           // with the currently logged in account, or by some DB lookup.
-          const auth = await findOrCreateAuthByAuthIdentity(provider.id, providerProfile.id, getUserFields);
+          const userId = await getOrCreateUserIdByOAuthProvider(provider.id, providerProfile.id, getUserFields);
+
+          const token = await createAuthToken(userId);
 
-          const token = await sign(auth.{= userFieldOnAuthEntityName =}.id);
           res.json({ token });
       })
     )
 
     return router;
 }
 
-async function findOrCreateAuthByAuthIdentity(
+async function getOrCreateUserIdByOAuthProvider(
```

Comment:
Well I really wish this function had a type signature for return type because then I could know now what it returns :D. 
But I don't understand what this function does. It gets or creates user id. How can it create user id? Does that mean that it creates the whole user? That sounds quite a big thing to do vs just get the user id.
What does it mean byOAuthProvider? We don't have this term much used around, do we? So this works only for 'google', 'github' and similar?

Sounds to me like this function is maybe doing too much and could certainly have a better name.

Reply:
Btw why is `providerName` not a union of string literals.

Reply:
I've now refactored this code a bit, this function no longer exists. But I'll refactor it a bit more since it feels weird now, even though it's a bit better.

Reply:
Ok, once you are done, let me know, and please point me to the new code so I can give it a look.

Reply:
Here's the updated code. I've tried to capture the flow:
- we need a user ID for auth token ...
- ... so, it's either a new or existing user
- ... (capture this logic in a function that at the end returns a user ID)
- ... use the function to get the user ID
- ... return the token

https://github.com/wasp-lang/wasp/pull/1583/files#diff-e79801968c3de90b3d94ff0d248e57b782a5b35e7b325989b4a6c4321998d332R57